guides/policy/azure-authenticator/index.html

```diff
@@ -0,0 +1,140 @@
+            <td>Application ID</td>
+            <td><input type="text" id="txtAppId"></td>
+            <td>The ID you want to use in the policy to identify this application.
+            Used as host ID.</td>
+        </tr>
+    </tbody>
+</table>
+
+<br />
+
+Azure access token
+<textarea class="container text-monospace small border" style="height: 291px" id="encodedToken"></textarea>
+
+<br />
+
+</p>
+<div class="show" id="collapseInfo">
+    <div class="card card-body">
+
+        <pre class="container border" id="theTemplate" hidden="hidden">
+      - !host
+        id: {{app ID}}
+        annotations:
+          authn-azure/subscription-id: {{subscription ID}}
+          authn-azure/resource-group: {{resource group}}
+          authn-azure/{{user or system}}-assigned-identity: {{identity}}
+</pre>
+
+        <h3>Policy</h3>
+        <pre class="container border" id="output">
+</pre>
+
+    </div>
+</div>
+
+<script>
+
+    $(document).ready(function () {
+
+        $('#encodedToken').on('input', function () {
+            updateCmd();
+        });
+
+        $('#tblInput input').on('input', function () {
+            updateCmd();
+        });
+    });
+
+    function updateCmd() {
+        let output = "";
+        let appId = $("#txtAppId").val();
+        let encodedToken = $("#encodedToken").val();
+        if (!appId){
+            output = "Application ID is missing";
+        }
+        else if (!encodedToken){
+            output = "Azure token is missing";
+        }
+        else {
+            output = getPolicyFromToken(encodedToken, appId);
+        }
+        $("#output").html(output);
+    }
+
+    function getPolicyFromToken(encodedToken, appId) {
+        try {
+            let policy = $("#theTemplate").html();
+            let parsedToken = parseJwt(encodedToken);
+            let token = parseAzureToken(parsedToken);
+
+            policy = policy.replace("{{app ID}}", appId);
+            policy = policy.replace("{{subscription ID}}", token.subscriptionId);
+            policy = policy.replace("{{resource group}}", token.resourceGroup);
+            policy = policy.replace("{{user or system}}", token.isUserIdentity ? "user" : "system");
+            policy = policy.replace("{{identity}}", token.identity);
+            return policy;
+        }
+        catch (e) {
+            console.error(e);
+            return "Failed to parse Azure Token"
+        }
+    }
+
+    function parseJwt (token) {
+        var base64Url = token.split('.')[1];
+        var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+        var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
+            return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+        }).join(''));
+
+        return JSON.parse(jsonPayload);
+    }
+
+    function parseAzureToken (token) {
+        var azureIdentity = {};
+
+        var xmsMirid = token.xms_mirid.split('/');
+        for (i = 0; i < xmsMirid.length; i++){
```

Comment:
I think it is better to mimic the parsing we use in the server. This way you lower the chance to be valid in this tool but fail in the server

Reply:
As @sigalsax mentioned, we do not depend on exact index in the xms_mirid parsing, so I can't use the [,,,] syntax.
I used `reduce` instead, what do you think of it now?